File: C#/AnimalsFight/Fighter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Game.AnimalsFight
{
    public class Fighter
    {
        string _name;
        int _power;

        public int Power { get => _power; }

        public Fighter(string name, int power)
        {
            _name = name;
            _power = power;

        }

        public void Introduce()
        {
            Console.WriteLine($"Hi I am {_name}");
        }
    }
}

File: Golang/animalsFight/fighter.go
package animalsfight

import "fmt"

type Fighter struct {
	name  string
	power int
}

func NewFighter(name string, power int) *Fighter {
	return &Fighter{name, power}
}

func (f *Fighter) Introduce() {
	fmt.Println("Hi I am " + f.name)
}
